shared/actions/tracker.js

```diff
@@ -12,45 +12,21 @@ import {isMobile} from '../constants/platform'
 import {type TypedState} from '../constants/reducer'
 import type {FriendshipUserInfo} from '../profile/friendships'
 
-// TODO sagaize this
-// Currently most things will call these thunks but they also tend to dispatch a generated action with the same name
-
-const startTimer = () => (dispatch: Dispatch, getState: () => TypedState) => {
-  // Increments timerActive as a count of open tracker popups.
-  dispatch(TrackerGen.createSetStartTimer())
-  const timerActive = getState().tracker.timerActive
-  if (timerActive === 1) {
-    // We're transitioning from 0->1, no tracker popups to one, start timer.
-    const intervalId = setInterval(() => {
-      const timerActive = getState().tracker.timerActive
-      if (timerActive <= 0) {
-        // All popups are closed now.
-        clearInterval(intervalId)
-      }
-
-      RPCTypes.trackCheckTrackingRpcPromise()
-    }, Constants.rpcUpdateTimerSeconds)
-  }
-}
-
-function setupUserChangedHandler() {
-  const setupUserChangedHandlerHelper = (dispatch: Dispatch, getState: () => TypedState) => {
-    engine().setIncomingHandler('keybase.1.NotifyUsers.userChanged', ({uid}) => {
-      dispatch(TrackerGen.createCacheIdentify({uid, goodTill: 0}))
-      const username = _getUsername(uid, getState())
-      if (username) {
-        dispatch(getProfile(username))
-      }
-    })
+// Send a heartbeat while trackers are still open
+function* _trackerTimer(): Generator<any, void, any> {
```

Comment:
instead of explicitly trying to bookkeep this and keeping track of timers we just spawn a saga which periodically checks and sends the rpc if there are any trackers alive